src/suggestion-builder.js

```diff
@@ -0,0 +1,66 @@
+import _ from 'lodash'
+import { buildClause } from './utils'
+
+export default function suggestionBuilder(newSuggestion) {
+    let suggestions = _.isEmpty(newSuggestion) ? {} : newSuggestion
+
+    function makeSuggestion(type, field, options) {
+        let suggestName
+        const { name, text } = options
+
+        if (name) {
+            suggestName = name
+            _.unset(options, 'name')
+        } else {
+            suggestName = `suggest_${type}_${field}`
+        }
+
+        if (text) {
+            _.unset(options, 'text')
+        }
+
+        const innerClause = Object.assign({}, {
+            text,
+            [type]: buildClause(field, null, options)
+        })
+
+        Object.assign(suggestions, {
+            [suggestName]: innerClause
+        })
+    }
+
+    return {
+        /**
+         * Add an suggestion clause to the query body.
+         *
+         * @param  {string}        field     Name of the field to suggest on.
+         * @param  {Object}        [options] (optional) Additional options to
+         *                                   include in the aggregation.
+         *                         [options.text ] text query to run on suggest
+         *                         [options.name ] pass a custom name to the function
+         *                         [options.analyzer ] name of predefined analyzer to use on suggest
+         * 
+         * @return {bodybuilder} Builder.
+         *
+         * @example
+         * bodybuilder()
+         *   .suggest('price', { text: 'test' })
+         *   .build()
+         *
+         * bodybuilder()
+         *   .suggest('price', { text: 'test', name: 'custom name' })
+         *   .build()
```

Comment:
In addition to these JSdocs can you also add a section to the readme showing usage of the suggest builder?

Reply:
Done! Let me know if you'd like any more docs. Also, are the docs on `bodybuilder.js.org` generated, or do I need to add those somewhere?

Reply:
Thanks! Those are generated and pushed to github pages when a new version is published, no more action needed on your part 👍